lib/water_drop/message.rb

```diff
@@ -1,17 +1,19 @@
 module WaterDrop
   # Message class which encapsulate single Kafka message logic and its delivery
   class Message
-    attr_reader :topic, :message
+    attr_reader :topic, :message, :partition, :partition_key
 
     # @param topic [String, Symbol] a topic to which we want to send a message
     # @param message [Object] any object that can be serialized to a JSON string or
     #   that can be casted to a string
     # @return [WaterDrop::Message] WaterDrop message instance
     # @example Creating a new message
     #   WaterDrop::Message.new(topic, message)
-    def initialize(topic, message)
+    def initialize(topic, message, options = {})
```

Comment:
This woulld be awesome. I thinkwe could leave topic, message, options = {} as you did and internally merge it into one hash that is forwarded into procuder.produce. If you are willing to do that, this would be more than awesome!